Задание №2: Grep

LDVSOFT opened the pull request.

LDVSOFT committed:
Added Grep command

LDVSOFT committed:
Splited Grep's process function:

Now processing one file is done in another function, `processFile',
`execute' now only processes options, arguments and IO.

yurii-litvinov reviewed (changes requested):
- Ещё хотелось описание, почему была выбрана именно такая библиотека
- Нужны юнит-тесты
- `echo 1 | grep 1` не работает

LDVSOFT committed:
package net.ldvsoft.spbau;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;

import static org.junit.Assert.assertEquals;

/**
 * Base test utils
 */
class ShellTestUtils {
    /**
     * Run shell with given input and test it's output
     * @param input input to give to the shell
     * @param expectedOutput expected output of the shell
     */
    static void doTest(String input, String expectedOutput) {
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
        Shell shell = new Shell(new ByteArrayInputStream(input.getBytes()), outputStream);
        shell.work();
        String actualOutput = outputStream.toString();
        assertEquals("Shell output is wrong!", expectedOutput, actualOutput);
    }
}

LDVSOFT committed:
Fixed Grep

LDVSOFT committed:
Added more tests for Grep

LDVSOFT committed:
Added more tests

LDVSOFT commented:
Добавил тестов, починил)

Либа -- потому что Apache Commons уже представляет собой хорошо проверенный набор стандартных библиотек, и поэтому как увидел, что она подходит, сразу взял.
